include/arch/structs.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) BayLibre SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * The purpose of this file is to provide essential/minimal architecture-
+ * specific structure definitions to be included in generic kernel
+ * structures.
+ *
+ * The following rules must be observed:
+ *  1. arch/structs.h shall not depend on kernel.h both directly and
+ *     indirectly (i.e. it shall not include any header files that include
+ *     kernel.h in their dependency chain).
+ *  2. kernel.h shall imply arch/structs.h via kernel_structs.h , such that
+ *     it shall not be necessary to include arch/structs.h explicitly when
+ *     kernel.h is included.
+ */
+
+#ifndef ZEPHYR_INCLUDE_ARCH_STRUCTS_H_
+#define ZEPHYR_INCLUDE_ARCH_STRUCTS_H_
+
+#if !defined(_ASMLANGUAGE)
+
+#if defined(CONFIG_ARM64)
+#include <arch/arm64/structs.h>
+#else
```

Comment:
This construction works fine I guess, but this is a persistent problem int the arch layer where someone will define an optional API and implement it for only a handful of platforms, so we end up with lots of #if'ery in contexts like this, or poorly standardized ARCH_HAS_FOO kconfigs, etc...

Seems like a layer that could automatically detect these things (maybe with a quick precompilation pass a-la autotools?) would be a good addition.